plugin/storage/grpc/proto/storage.proto

```diff
@@ -0,0 +1,201 @@
+        StoragePluginError error = 2;
+    }
+}
+
+message WriteSpanRequest {
+    Span span = 1;
+}
+
+message WriteSpanResponse {
+    oneof response {
+        EmptyResponse success = 1;
+        StoragePluginError error = 2;
+    }
+}
+
+message GetTraceRequest {
+    bytes trace_id = 1 [
+     (gogoproto.nullable) = false,
+     (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+     (gogoproto.customname) = "TraceID"
+    ];
+}
+
+message GetTraceSuccess {
+    Trace trace = 1;
+}
+
+message GetTraceResponse {
+    oneof response {
+        GetTraceSuccess success = 1;
+        StoragePluginError error = 2;
+    }
+}
+
+message GetServicesRequest {}
+
+message GetServicesSuccess {
+    repeated string services = 1;
+}
+
+message GetServicesResponse {
+    oneof response {
+        GetServicesSuccess success = 1;
+        StoragePluginError error = 2;
+    }
+}
+
+message GetOperationsRequest {
+    string service = 1;
+}
+
+message GetOperationsSuccess {
+    repeated string operations = 1;
+}
+
+message GetOperationsResponse {
+    oneof response {
+        GetOperationsSuccess success = 1;
+        StoragePluginError error = 2;
+    }
+}
+
+message TraceQueryParameters {
+    string service_name = 1;
+    string operation_name = 2;
+    map<string, string> tags = 3;
+    google.protobuf.Timestamp start_time_min = 4 [
+     (gogoproto.stdtime) = true,
+     (gogoproto.nullable) = false
+    ];
+    google.protobuf.Timestamp start_time_max = 5 [
+     (gogoproto.stdtime) = true,
+     (gogoproto.nullable) = false
+    ];
+    google.protobuf.Duration duration_min = 6 [
+     (gogoproto.stdduration) = true,
+     (gogoproto.nullable) = false
+    ];
+    google.protobuf.Duration duration_max = 7 [
+     (gogoproto.stdduration) = true,
+     (gogoproto.nullable) = false
+    ];
+    int32 num_traces = 8;
+}
+
+message FindTracesRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTracesSuccess {
+    repeated Trace traces = 1;
+}
+
+message FindTracesResponse {
+    oneof response {
+        FindTracesSuccess success = 1;
+        StoragePluginError error = 2;
+    }
+}
+
+message FindTraceIDsRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsSuccess {
+    repeated bytes trace_id = 1 [
+     (gogoproto.nullable) = false,
+     (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+     (gogoproto.customname) = "TraceIDs"
+    ];
+}
+
+message FindTraceIDsResponse {
+    oneof response {
+        FindTraceIDsSuccess success = 1;
+        StoragePluginError error = 2;
+    }
+}
+
+message EmptyResponse {
+
+}
+
+message StoragePluginError {
+    string message = 1;
+}
+
+service StoragePlugin {
+    // dependencystore/Writer
+//    rpc WriteDependencies(WriteDependenciesRequest) returns (WriteDependenciesResponse);
+    // dependencystore/Reader
+    rpc GetDependencies(GetDependenciesRequest) returns (GetDependenciesResponse);
```

Comment:
Is there a strong reason for merging all different types of storage into a single proto `service`? Could we not have the same component implement different services?